js/ez-selection-activate-element.js

```diff
@@ -0,0 +1,122 @@
+     * `selection-selector` attribute allows to restrict this behavior to a given
+     * subset of checkboxes (or to radio buttons for instance).
+     *
+     * Examples:
+     *
+     * Enable a button based on a checkboxes:
+     * ```
+     * <ez-selection-activate-element>
+     *     <ul>
+     *         <li><input type="checkbox"></li>
+     *         <li><input type="checkbox"></li>
+     *     </ul>
+     *     <button class="ez-js-activable-element" disabled>Do!</button>
+     * </ez-selection-activate-element>
+     * ```
+     *
+     * Enable a button only if one element is selected:
+     * ```
+     * <ez-selection-activate-element>
+     *     <ul>
+     *         <li><input type="checkbox"></li>
+     *         <li><input type="checkbox"></li>
+     *     </ul>
+     *     <button class="ez-js-activable-element ez-js-activable-element-single-selection" disabled>Do!</button>
+     * </ez-selection-activate-element>
+     * ```
+     *
+     * Enable a button based on a subset of checkboxes:
+     * ```
+     * <ez-selection-activate-element selection-selector=".checkboxes-that-matches">
+     *     <ul>
+     *         <li><input type="checkbox" class"this-checkbox-will-not-match"></li>
+     *         <li><input type="checkbox" class"checkboxes-that-matches"></li>
+     *     </ul>
+     *     <button class="ez-js-activable-element" disabled>Do!</button>
+     * </ez-selection-activate-element>
+     * ```
+     *
+     * Among others standard APIs, this component relies on `Element.matches`.
+     * `Element.matches` is not available in Edge 14. So for this component to
+     * work in those browsers, the page should include polyfills of those
+     * standard API.
+     *
+     * @polymerElement
+     * @demo demo/ez-selection-activate-element.html
+     */
+    class SelectionActivateElement extends Polymer.Element {
+        static get is() {
+            return 'ez-selection-activate-element';
+        }
+
+        static get properties() {
+            return {
+                /**
+                 * Holds a selector for checkbox/radio inputs to track.
+                 */
+                selectionSelector: {
+                    type: String,
+                    value: 'input[type="checkbox"]',
+                },
+            };
+        }
+
+        connectedCallback() {
+            super.connectedCallback();
+            this.addEventListener('click', this._updateActivableElements.bind(this));
+        }
+
+        /**
+         * Enables or disables the activable element depending on the checked
+         * checkboxes or radio buttons.
+         *
+         * @param {Event} e
+         */
+        _updateActivableElements(e) {
+            if (e.target.matches(this.selectionSelector)) {
+                const activableElements = this._getActivableElements();
+                const checkedElementsNumber = this._getCheckedElementsNumber();
+
+                activableElements.forEach((element) => {
```

Comment:
I have a question about Edge. As far as I know, `forEach` is not available on {NodeList} in Edge 14. Do we support Edge 14? I ask because you advice to use `matches` polyfill for Edge 14. (Full support for `matches` is in Edge 15).